src/amber/scripts/environment_loader.cr

```diff
@@ -1,75 +0,0 @@
-require "yaml"
-require "secure_random"
-require "../support/message_encryptor"
-
-environment = ARGV[0]? || ENV["AMBER_ENV"]? || "development"
-
-# Override to make the code testable.
-# Eliminates the need for environment variables gettings set before tests
-env_path = begin
-  if File.exists?(fn = "./config/environments")
-    fn
-  else # if config/environments doesn't exist look in spec folder and set env to test
-    environment = ARGV[0]? || "test"
-    "./spec/support/config"
-  end
-end
-
-secret_key = ENV["AMBER_SECRET_KEY"]? || begin
-  File.open(".amber_secret_key").gets_to_end.to_s if File.exists?(".amber_secret_key")
-end
-
-yml = if File.exists?(fn = "#{env_path}/#{environment}.yml")
-        File.read(fn)
-      elsif File.exists?(fn = "#{env_path}/.#{environment}.enc") && secret_key
-        enc = Amber::Support::MessageEncryptor.new(secret_key)
-        String.new(enc.decrypt(File.open(fn).gets_to_end.to_slice))
-      else
-        "env: #{environment}"
-      end
-
-settings = YAML.parse(yml)
-
-str = String.build do |s|
-  # Most of this logic can be cleaned up by just requiring environment files to contain valid params.
-  # For now I have this here so that tests can still pass without having to load env files although they should.
-  # This is a transistion.
-  s.puts %(@name = "#{settings["name"]? || "Amber_App"}")
```

Comment:
So much red. I agree this is for the best but it makes me sad.

Reply:
Don't be, most of the inspiration came from your version 😄